### DISCUSSION TITLE
P2P Jamulus fork

### DISCUSSION THREAD

**Discussion by ann0see** [jamulussoftware/jamulus#2016] [Ideas]
While doing research, I found a fork of Jamulus implementing P2P.
I know P2P came up some while ago, therefore some of you might be interested in this project by @InEs-HPMM https://github.com/InES-HPMM/jamulus_direct which seems to be a modified version of Jamulus but with peer to peer connection.

CC: @JanaUnderground who seems to be a person with push access

**Comment by ghost** [comment]
Jamulus already has peer to peer capability:
The person that runs both a local client and server has a reciprocal peer to peer connection to everyone else. Everyone else to everyone else has a star network connection to the server.

**Comment by JanaUnderground** [comment]
Hi,

@DavidSavinkoff: Correct, but only one person has a P2P connection. With jamulus_direct, we wanted peer-to-peer connections between all clients instead of the star topology.

Basically, we left the server structure but added an option for each client to send its audio P2P to each other client connected to the server. In a session with more than two participants, clients that are not running a server then have a much smaller network delay between each other since the audio does not have to do the trip over the server. 

In a session with two participants, where one runs the server locally, Jamulus already is P2P.
